lib/inventory/inventory_api.rb

```diff
@@ -348,6 +545,41 @@ def to_h
     end
   end
 
+  class MetricType
+    # @return [String] Full path of the metric (type)
+    attr_reader :path
+    # @return [String] Name of the metric
+    attr_reader :name
+    attr_reader :id
+    attr_reader :feed
+    attr_reader :unit
+    # @return [Long] collection interval in seconds
+    attr_reader :collection_interval
+
+    def initialize(type_hash)
+      @id = type_hash['id']
+      @path = type_hash['path']
+      @name = type_hash['name'] || @id
+      @_hash = type_hash.dup
+
+      tmp = path.split('/')
+      tmp.each do |pair|
+        (key, val) = pair.split(';')
```

Comment:
I might be wrong here, but I think `String#split`  receives a `Regexp` by default. So, you could use `/;/` instead of letting it accept a string and transform into a Regexp.